src/ripple/core/JobCoro.h

```diff
@@ -39,10 +42,14 @@ struct JobCoro_create_t { };
 class JobCoro : public std::enable_shared_from_this<JobCoro>
 {
 private:
+    detail::LocalValues lvs_;
     JobQueue& jq_;
     JobType type_;
     std::string name_;
+    bool running_;
     std::mutex mutex_;
+    std::mutex mutex_run_;
```

Comment:
Damn, you're right. I thought`mutex_` is used for the condition variable and `mutex_run_` was just used to protect `running_`.